modules/s4/app.tf

```diff
@@ -0,0 +1,198 @@
+    update = "1h"
+  }
+  type = var.disk_type == "hyperdisk-extreme" ? "pd-ssd" : var.disk_type
+  zone = var.zone1_name
+}
+
+resource "google_compute_disk" "sapdapp11_usr_sap" {
+  count = var.app_vms_multiplier
+  lifecycle {
+    ignore_changes = [snapshot]
+  }
+  name    = "${var.vm_prefix}app1${1 + (count.index * 2)}-usr-sap"
+  project = data.google_project.sap-project.project_id
+  size    = var.app_disk_usr_sap_size
+  timeouts {
+    create = "1h"
+    delete = "1h"
+    update = "1h"
+  }
+  type = var.disk_type == "hyperdisk-extreme" ? "pd-ssd" : var.disk_type
+  zone = var.zone1_name
+}
+
+resource "google_compute_instance" "sapdapp11" {
+  allow_stopping_for_update = var.allow_stopping_for_update
+  attached_disk {
+    device_name = google_compute_disk.sapdapp11_usr_sap[count.index].name
+    source      = google_compute_disk.sapdapp11_usr_sap[count.index].self_link
+  }
+  attached_disk {
+    device_name = google_compute_disk.sapdapp11_export_interfaces[count.index].name
+    source      = google_compute_disk.sapdapp11_export_interfaces[count.index].self_link
+  }
+  boot_disk {
+    auto_delete = false
+    device_name = "persistent-disk-0"
+    source      = google_compute_disk.sapdapp11[count.index].self_link
+  }
+  count = var.app_vms_multiplier
+  lifecycle {
+    ignore_changes = [
+      min_cpu_platform,
+      network_interface[0].alias_ip_range,
+      metadata["ssh-keys"]
+    ]
+  }
+  machine_type = var.app_machine_type
+  metadata = {
+    enable-oslogin = "FALSE"
+    ssh-keys       = ""
+  }
+  min_cpu_platform = lookup(local.cpu_platform_map, var.app_machine_type, "Automatic")
+  name             = "${var.vm_prefix}app1${1 + (count.index * 2)}"
+  network_interface {
+    dynamic "access_config" {
+      content {
+      }
+      for_each = var.public_ip ? [1] : []
+    }
+    network    = data.google_compute_network.sap-vpc.self_link
+    network_ip = google_compute_address.sapdapp11-1[count.index].address
+    subnetwork = data.google_compute_subnetwork.sap-subnet-app-1.self_link
+  }
+  project = data.google_project.sap-project.project_id
+  scheduling {
+    automatic_restart   = true
+    on_host_maintenance = "MIGRATE"
+    preemptible         = false
+  }
+  service_account {
+    email  = data.google_service_account.service_account_app.email
+    scopes = ["https://www.googleapis.com/auth/cloud-platform"]
+  }
+  tags = ["${var.deployment_name}-s4-comms"]
+  zone = var.zone1_name
+}
+
+resource "google_dns_record_set" "to_vm_sapdapp11" {
+  count        = var.app_vms_multiplier
+  managed_zone = data.google_dns_managed_zone.sap_zone.name
+  name         = "${var.vm_prefix}app1${1 + (count.index * 2)}.${data.google_dns_managed_zone.sap_zone.dns_name}"
+  project      = data.google_project.sap-project.project_id
+  rrdatas = [
+    google_compute_instance.sapdapp11[count.index].network_interface[0].network_ip
+  ]
+  ttl  = 300
+  type = "A"
+}
+
+resource "google_project_iam_member" "app_sa_role_1" {
```

Comment:
At the moment making this change in TF would be difficult due to the build processes we use. So this could be useful in the future but for now can't be done.